### DISCUSSION TITLE
Deploy without App Registrations

### DISCUSSION THREAD

**Discussion by plazovnikACN** [Azure/ipam#142] [Q&A]
Hey guys,

is it possible to deploy IPAM without the super permissions

Like, what needs to be done for [App Registration Only Deployment](https://azure.github.io/ipam/#/deployment/README?id=app-registration-only-deployment), if I want to do it manually?

And then only run [Infrastructure Stack (Only) Deployment](https://azure.github.io/ipam/#/deployment/README?id=infrastructure-stack-only-deployment) with a subscription admin?

**Comment by DCMattyG** [comment]
Hello @plazovnikACN, appreciate you reaching out on this!

The reason that a higher level of permissions is needed is because there are certain processes that need to be done which require an elevated level of permissions. For instance, someone needs to grant [Admin Consent](https://learn.microsoft.com/en-us/azure/active-directory/manage-apps/grant-admin-consent) to the App Registrations, so even if you were to manually create them, this will still need to be done.

That said, we understand that many times within an organization, the team whom would create/approve the App Registration and the team whom would deploy the Azure Infrastructure are *not the same*. To address this, we have a 2-part deployment method where whatever team control Azure AD can deploy the App Registrations on your behalf, then pass a generated Parameters file to the infrastructure team which they can use to deploy the actual Azure resources.

I see you've already pointed this out above, but I'll put the links here anyway for completeness...

Step 1 (App-Registration Deployment):
https://azure.github.io/ipam/#/deployment/README?id=app-registration-only-deployment

Step 2 (Infrastructure Deployment):
https://azure.github.io/ipam/#/deployment/README?id=infrastructure-stack-only-deployment

So going back to your original question, I'm not sure what you'd gain by doing this manually as you would *STILL* require elevated permissions to be consented to in order for IPAM to function properly, so whether you click everything manually in the Azure Portal, or run the script, the exact same permissions will be needed.

All of that said, is there some resistance from your Azure AD team which would prevent you from deploying the Azure IPAM solution, or were you just exploring if there was a mechanism where you could deploy this without going through that team (if possible)?

We're here to help you however we can, so let us know where your stuck....we'll work through it together!
